plugins/scaffolder/src/components/MultistepJsonForm/MultistepJsonForm.tsx

```diff
@@ -63,6 +63,15 @@ export const MultistepJsonForm = ({
     setActiveStep(Math.min(activeStep + 1, steps.length));
   const handleBack = () => setActiveStep(Math.max(activeStep - 1, 0));
 
+  const transformErrors = (errors: AjvError[]) =>
+    errors.map(error => {
+      if (error.name === 'pattern') {
+        error.message = 'does not match the required pattern';
```

Comment:
I wonder do you get the desired pattern here? Would be good to show in the form maybe as the audience should understand the `RegEx` too

Reply:
If we want the error to include the RegEx pattern, I can just remove `transformErrors()` and let react-jsonschema-form do its default behaviour. Then the message shown will be _".component-id does not match pattern "^[a-z0-9]+(-[a-z0-9]+)*$"_. Does that sound good? @benjdlambert

Reply:
@spotify/backstage-core thoughts?

Reply:
Oh tough question. As a dev I like the regex version, but realize that that doesn't apply to everyone. If we can't have per field custom written details, I still think we can keep the original regex version.

Reply:
yeah, one idea I had was to add something like an `error_message` key to the property in template.yaml like so: 
<img width="524" alt="Screenshot 2020-10-27 at 14 57 20" src="https://user-images.githubusercontent.com/16206098/97311922-3121e300-1865-11eb-8f53-aa306da82461.png">

We can then use the `error_msg` in `transformErrors()` to display a nicer, per-field details. It's trivial to implement but I'm not sure how good of an idea it is? @freben @benjdlambert

Reply:
I thought I actually recalled that this was a feature in json schema forms that you could define the error message in the schema, but a quick google suggests that it might not be available and it was just a dream. I think for now, let's remove the transformErrors part and we can revisit if there's a need in the future